### DISCUSSION TITLE
BLE Device

### DISCUSSION THREAD

**Discussion by jmgech** [h2zero/NimBLE-Arduino#710] [General]
Hi, I'm just discovering Nimble for a small project, and am facing some questions :)

The goal is to configure an ESP32 as a gamepad, to which a computer or mobile device will connect to and get data back.

At this stage, I'm able to:
- Make an initial bind and connection
- Get data back from a python script

On the Nimble side I'm noticing that if I manually disconnect from the device, it immediately auto reconnects. I made sure to remove all restart advertising code I had, but it's still occurring and I don't know why.

I'm noticing that subsequent use of the script on a bound device fails. If I forget the device, and rebind it, the script works again (until a subsequent disconnection...). This may be due to python code ( and Bleak). It seems that I'm able to see the device if it's unbound and unconnected, but as soon as it's bound, I can no longer read from it.

I would appreciate any feedback or help on this !

**Arduino script:**
[main.txt](https://github.com/user-attachments/files/16643650/main.txt)

**Python Bleak** Updated
[bleak.txt](https://github.com/user-attachments/files/16643651/bleak.txt)

Thanks!
J.

**Comment by jmgech** [comment]
This seems to be linked to NimBLESecurity. If I replace my sample NimBLESecurity code with the following, the python code works fine, even after disconnecting / reconnecting:

    /*NimBLESecurity* pSecurity = new NimBLESecurity();
    pSecurity->setAuthenticationMode(ESP_LE_AUTH_REQ_SC_MITM_BOND);  // Require bonding and MITM protection
    pSecurity->setCapability(ESP_IO_CAP_IO);  // Require a passkey
    pSecurity->setInitEncryptionKey(ESP_BLE_ENC_KEY_MASK | ESP_BLE_ID_KEY_MASK);  // Enable encryption
    */
   // Temporarily disable security for testing
    NimBLESecurity* pSecurity = new NimBLESecurity();
    pSecurity->setAuthenticationMode(ESP_LE_AUTH_NO_BOND);  // No bonding required

**Reply by h2zero** [reply]
Have you set the server advertise on disconnect to false?
